src/ol/renderer/webgl/TileLayer.js

```diff
@@ -468,23 +472,17 @@ class WebGLTileLayerRenderer extends WebGLLayerRenderer {
     }
 
     // TODO: let the renderers manage their own cache instead of managing the source cache
-    if (tileSource.canExpireCache()) {
-      /**
-       * @param {import("../../PluggableMap.js").default} map Map.
-       * @param {import("../../PluggableMap.js").FrameState} frameState Frame state.
-       */
-      const postRenderFunction = function (map, frameState) {
-        const tileSourceKey = getUid(tileSource);
-        if (tileSourceKey in frameState.usedTiles) {
```

Comment:
This was never true since the renderer never adds the tile source key to the `usedTiles` lookup.  Things would be simpler if we removed the tile cache from sources altogether (and instead maintained them on renderers).

Reply:
Thanks for fixing this. I agree it would simplify things to let the renderer cache tiles, instead of the source. Definitely something to keep in mind for a refactoring in the near future.